packages/flutter_test/lib/src/binding.dart

```diff
@@ -700,13 +729,33 @@ abstract class TestWidgetsFlutterBinding extends BindingBase
     // The LiveTestWidgetsFlutterBinding overrides this to report the exception to the console.
   }
 
+  Future<dynamic> _handleMessage(dynamic mockMessage) async {
+    final Map<dynamic, dynamic> data =
```

Comment:
This method should check for type mockMessage['type'] before accessing the data.